parsec/core/fs/remote_loader.py

```diff
@@ -329,6 +329,45 @@ def __init__(
         # TODO: also store access id in remote_manifest and check it here
         return remote_manifest
 
+    async def list_versions(self, entry_id: EntryID) -> Dict:
+        """
+        Raises:
+            FSError
+            FSBackendOfflineError
+            FSWorkspaceInMaintenance
+            FSRemoteManifestNotFound
+        """
+        try:
+            versions_dict = await self.backend_cmds.vlob_list_versions(entry_id)
+
+        # Vlob is not found
+        except BackendCmdsNotFound as exc:
+            raise FSRemoteManifestNotFound(entry_id) from exc
+
+        # Backend is not available
+        except BackendNotAvailable as exc:
+            raise FSBackendOfflineError(str(exc)) from exc
+
+        # Workspace is in maintenance
+        except BackendCmdsInMaintenance as exc:
+            raise FSWorkspaceInMaintenance(
+                f"Cannot download vlob while the workspace is in maintenance"
+            ) from exc
+
+        except BackendCmdsBadResponse as exc:
+            if exc.status == "not_found":
+                raise FSRemoteManifestNotFound(entry_id)
+            else:
+                raise FSError(f"Cannot fetch vlob {entry_id}: {exc}") from exc
+
+        # Another backend error
+        except BackendConnectionError as exc:
+            raise FSError(f"Cannot fetch vlob {entry_id}: {exc}") from exc
+
+        # author = await self.remote_device_manager.get_device(expected_author_id)
```

Comment:
Code mort à virer ?

Reply:
En fait, j'hésitais, et j'ai oublié de te demander : est-ce que ca ne serait pas intéressant d'itérer le dict et de remplacer les DeviceID par des VerifiedRemoteDevice directement pendant cet appel? Sachant que le remote_devices_manager a très probablement les devices en cache, ça permettrait par exemple d'accéder directement aux noms des utilisateurs avec la structure retournée par cet appel

Reply:
>  est-ce que ca ne serait pas intéressant d'itérer le dict et de remplacer les DeviceID par des VerifiedRemoteDevice directement pendant cet appel?

1) J'ai l'impression qu'on risque d'induit en erreur l'utilisateur en faisant ça car `list_versions` est une api qui fait confiance au backend (cad le backend peut décider de nous envoyer des infos arbitraires sans qu'on puisse s'en rendre compte), du coup retourner des `VerifiedRemoteDevice` qui eux sont vérifiés donne une fausse impression de sécurité...
2) c'est pas très couteux de faire une boucle pour récupérer les `VerifiedRemoteDevice` à partir des DeviceID qui nous ont été retournés, donc autant ne pas le faire dans `list_versions` pour éviter de le faire pour rien

> ça permettrait par exemple d'accéder directement aux noms des utilisateurs avec la structure retournée par cet appel

On peut directement faire `device_id.user_id` pour ça, ou alors tu penses à un autre champ ?

Reply:
Effectivement, j'avais donné un exmple sans trop réfléchir, dans la très grande majorité des cas de DeviceID suffit, je laisse tel quel